packages/client/hmi-client/src/workflow/ops/model/tera-model-node.vue

```diff
@@ -88,11 +93,37 @@ async function onModelChange(chosenProjectModel: ProjectAsset) {
 	await getModelById(chosenProjectModel.assetId);
 }
 
+// Create a preview image based on MMT
+const generateTemplatePreview = async () => {
+	if (!model.value) return;
+	try {
+		const kernelManager = new KernelSessionManager();
+		await kernelManager.init('beaker_kernel', 'Beaker Kernel', {
+			context: 'mira_model',
+			language: 'python3',
+			context_info: {
+				id: model.value.id
+			}
+		});
+
+		kernelManager
+			.sendMessage('reset_request', {})
+			.register('reset_response', () => null) // noop
+			.register('model_preview', (data) => {
+				templatePreview.value = `data:image/png;base64, ${data?.content?.['image/png']}`;
+			});
+	} catch (err) {
+		console.error(err);
+	}
+};
+
 onMounted(async () => {
 	const state = props.node.state;
 	if (state.modelId) {
 		model.value = await getModel(state.modelId);
 
+		await generateTemplatePreview();
```

Comment:
Do we need the `await` here, I do not think we need to wait for it to keep doing logic.